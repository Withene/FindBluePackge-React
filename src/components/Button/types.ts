import React from 'react';
export type ButtonStyleProps = {
  Center: boolean;
  BorderColor?: string;
  MBottom?: boolean;
};

export interface ButtonProp extends React.HTMLAttributes<HTMLElement> {
  Text: string;
  Center: boolean;
  eventClick?: (e?: React.MouseEvent) => void | Promise<unknown>;
  type?: 'submit';
  disabled?: boolean;
  Name?: string;
  ColorBorder?: string;
  MBottom?: boolean;
}
