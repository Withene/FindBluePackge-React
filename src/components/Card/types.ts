export type CardProps = {
  FirtsTitleProp: string;
  OrigemProp?: string;
  DestinoProp?: string;
  Time: string;
  LocalProp?: string;
};
