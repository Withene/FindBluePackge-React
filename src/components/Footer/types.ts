export interface FooterProps {
  height?: string;
}
