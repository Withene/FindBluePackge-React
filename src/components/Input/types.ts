import React from 'react';
export interface InputProp extends React.HTMLAttributes<HTMLElement> {
  Text?: string;
  inputType: string;
  event?: (event: any) => void;
  eventBlur?: (event: any) => void;
  value?: string;
  name: string;
  maxwidth?: string;
  Mbottom?: string;
}

export interface CoverProps {
  maxwidth?: string;
  Mbottom?: string;
}

//ONLY FOR TESTS BY JEST

export interface InputTestProps extends HTMLElement {
  value?: string;
}
