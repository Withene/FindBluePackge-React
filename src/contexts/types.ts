import { ReactNode } from 'react';

export type LoadResponseProvider = {
  children: ReactNode;
};

export type AuthContextType = {
  handleLoadPackge: (data: DataProps) => Promise<void>;
  PackDates: Array<PackDatesProps>;
  have: boolean;
};

export type DataProps = Array<string>;

export type PackDatesProps = {
  status?: string;
  data?: string;
  origem?: string;
  hora?: string;
  local?: string;
};
