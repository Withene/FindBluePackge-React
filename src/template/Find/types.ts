export type TypeOfValueState = {
  find: string;
};
export type TypesOfErros = {
  find?: string;
};
