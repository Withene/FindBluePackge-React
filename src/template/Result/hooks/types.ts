export type Result = {
  hora: string;
  status: string;
  origem: string;
  destino: string;
  local: string;
};
