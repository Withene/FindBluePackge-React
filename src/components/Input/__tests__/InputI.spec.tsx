import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Input } from '../index';
import { InputTestProps } from '../types';
import '@testing-library/jest-dom';

describe('Input', () => {
  it('should have a value of name and type', () => {
    const fn = jest.fn();
    render(<Input event={fn} placeholder="Place Holderhere" name="test" inputType={'text'} value={'test'} />);
    const input: InputTestProps = screen.getByPlaceholderText(/Place Holderhere/i);
    expect(input).toHaveAttribute('type', 'text');
    expect(input).toHaveAttribute('name', 'test');
  });

  it('should call handleChange function on each key pressed', () => {
    const value = 'o valor';
    const fn = jest.fn();
    render(
      <Input
        event={fn}
        placeholder="Place Holderhere"
        name="test"
        inputType={'text'}
        value={value}
        maxwidth={'22'}
        Mbottom={'2'}
      />,
    );

    const input: InputTestProps = screen.getByPlaceholderText(/Place Holderhere/i);

    userEvent.type(input, value);

    expect(input.value).toBe(value);

    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();

    const { container } = render(
      <Input event={fn} placeholder="Place Holderhere" name="test" inputType={'text'} value={'test'} />,
    );
    expect(container).toMatchSnapshot();
  });
});
