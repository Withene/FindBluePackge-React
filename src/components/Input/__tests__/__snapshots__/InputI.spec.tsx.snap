// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<div>
  <div
    class="sc-dlfnuX ghPOrW"
  >
    <label
      class="sc-gsTEea QIuXs"
    />
    <input
      class="sc-bdfBQB ehPXqP"
      name="test"
      placeholder="Place Holderhere"
      type="text"
      value="test"
    />
  </div>
</div>
`;
